### ISSUE TITLE
Change both local and global waypoint reached to use the line-based method

### ISSUE THREAD

**Issue by tylerlum** [UBCSailbot/local-pathfinding#171]
Task 1: Change Path waypointReached so that there is some distance between the waypoint and the line. (hard to word, look at examples below)

https://github.com/UBCSailbot/local-pathfinding/blob/master/python/Path.py#L341

WHAT WE WANT:
```
        Examples: B is boat. 0 is (self._nextWaypointIndex - 1). 1 is (self._nextWaypointIndex).

        Example 1 Waypoint not reached:
              |
              |
        0  B    1
              |
              |

        Example 2 Waypoint reached:
              |
              |
        0       1
              |
              |B
```

WHAT IT IS NOW:
```
        Examples: B is boat. 0 is (self._nextWaypointIndex - 1). 1 is (self._nextWaypointIndex).

        Example 1 Waypoint not reached:
                |
                |
        0  B    1
                |
                |

        Example 2 Waypoint reached:
                |
                |
        0       1
                |
                |B
```

Task 2: Replace globalWaypointReached (radius based) to be the same offset line-based approach as local waypoint reached.

https://github.com/UBCSailbot/local-pathfinding/blob/master/python/utilities.py#L130
^Likely remove this entirely

https://github.com/UBCSailbot/local-pathfinding/blob/master/python/main_loop.py#L115
^ likely replace

```
        isGlobalWaypointReached = utils.globalWaypointReached(state.position, state.globalWaypoint)
```

with 
```
        isGlobalWaypointReached = localPath.lastWaypointReached(state.position)
```

**Comment by tylerlum** [comment]
![image](https://user-images.githubusercontent.com/26510814/101868996-4f9d1e80-3b33-11eb-8642-d8571c30508d.png)

**Comment by tylerlum** [comment]
Resolved in #172
